File: internal/indexer/testdata/go/defs_c.go
package sample

import "models"

// CriticalUserModelCache caches CriticalUserModel instances.
type CriticalUserModelCache struct {
	entries map[int]models.CriticalUserModel
}

// Get retrieves a CriticalUserModel from the cache.
func (c *CriticalUserModelCache) Get(id int) (models.CriticalUserModel, bool) {
	m, ok := c.entries[id]
	return m, ok
}

// Put stores a CriticalUserModel in the cache.
func (c *CriticalUserModelCache) Put(id int, m models.CriticalUserModel) {
	if c.entries == nil {
		c.entries = make(map[int]models.CriticalUserModel)
	}
	c.entries[id] = m
}

File: internal/indexer/testdata/rust/defs_c.rs
use crate::models::critical_user_model::CriticalUserModel;

/// CriticalUserModelCache caches CriticalUserModel instances by id.
pub struct CriticalUserModelCache {
    entries: Vec<(u64, CriticalUserModel)>,
}

impl CriticalUserModelCache {
    pub fn new() -> CriticalUserModelCache {
        CriticalUserModelCache { entries: Vec::new() }
    }

    /// Inserts a CriticalUserModel with the given id.
    pub fn insert(&mut self, id: u64, model: CriticalUserModel) {
        self.entries.push((id, model));
    }

    /// Looks up a CriticalUserModel by id.
    pub fn get(&self, id: u64) -> Option<&CriticalUserModel> {
        self.entries.iter().find(|(k, _)| *k == id).map(|(_, v)| v)
    }
}
